core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Manager.java

```diff
@@ -154,6 +154,39 @@ public OAuth2Util.AuthSession tableSession(
         (OAuth2Util.AuthSession) parent);
   }
 
+  @Override
+  public AuthSession tableSession(
+      RESTClient sharedClient, Map<String, String> properties) {
+
+    AuthConfig config = AuthConfig.fromProperties(properties);
+    Map<String, String> headers = OAuth2Util.authHeaders(config.token());
+    OAuth2Util.AuthSession parent = new OAuth2Util.AuthSession(headers, config);
+
+    keepRefreshed(config.keepRefreshed());
+
+    // Important: this method is invoked from standalone components; we must not assume that
+    // the refresh client and session cache have been initialized, because catalogSession()
+    // won't be called.
+    if (refreshClient == null) {
+      refreshClient = sharedClient.withAuthSession(parent);
+    }
+    if (sessionCache == null) {
+      sessionCache = newSessionCache(name, properties);
+    }
+
+    if (config.token() != null) {
```

Comment:
This is basically the same logic as in `catalogSession()`, but with caching enabled.